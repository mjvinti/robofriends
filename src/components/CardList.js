import React from 'react'
import Card from './Card';

const CardList = ({ robots }) => {
    console.log('CardList');

    return (
        <div>
            {
                robots.map(({ id, name, email }) => {
                    return (
                        <Card
                            key={id}
                            id={id}
                            name={name}
                            email={email}
                        />
                    );
                })
            }
        </div>
    );
}

export default CardList;