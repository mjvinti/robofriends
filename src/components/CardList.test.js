import React from 'react';
import { shallow } from 'enzyme';

import CardList from './CardList';

it('should render CardList component', () => {
    const mockRobots = [{ id: 1, name: 'John Snow', email: 'john@snow.com' }];
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});