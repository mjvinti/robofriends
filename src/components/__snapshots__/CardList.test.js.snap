// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardList component 1`] = `
<div>
  <Card
    email="john@snow.com"
    id={1}
    key="1"
    name="John Snow"
  />
</div>
`;
